showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITGdch.java

```diff
@@ -0,0 +1,196 @@
+  private String projectId;
+  private URI tokenUri;
+
+  @Before
+  public void setup() throws IOException, URISyntaxException {
+    transportFactory = new MockTokenServerTransportFactory();
+    prepareCredentials();
+    tempFolder.create();
+    settings =
+        EchoSettings.newBuilder()
+            .setCredentialsProvider(FixedCredentialsProvider.create(initialCredentials))
+            .build();
+  }
+
+  @After
+  public void tearDown() {
+    if (client != null) {
+      client.close();
+    }
+  }
+
+  private void prepareCredentials() throws IOException, URISyntaxException {
+    // compute absolute path of the CA certificate
+    Path caCertPath = Paths.get(getClass().getResource(CA_CERT_RESOURCE_PATH).toURI());
+
+    // open gdch credential json (still needs its "ca_cert_path" to point to the CA certificate
+    // obtained from above)
+    JsonFactory factory = new GsonFactory();
+    GenericJson converted =
+        factory.fromInputStream(
+            getClass().getResourceAsStream(TEST_GDCH_CREDENTIAL_FILE), GenericJson.class);
+
+    // modify and save to a temporary folder
+    converted.set(CA_CERT_JSON_KEY, caCertPath.toAbsolutePath().toString());
+    projectId = converted.get("project").toString();
+    tokenUri = URI.create(converted.get("token_uri").toString());
+
+    File tempGdchCredentialFile = tempFolder.newFile(TEMP_CREDENTIAL_JSON_FILENAME);
+    try (FileWriter fileWriter = new FileWriter(tempGdchCredentialFile)) {
+      String preparedJson = converted.toPrettyString();
+      fileWriter.write(preparedJson);
+    }
+
+    // use temp location to instantiate credentials
+    initialCredentials = GdchCredentialsTestUtil.fromJson(converted, transportFactory);
+  }
+
+  /**
+   * {@link com.google.api.gax.rpc.ClientContext} will create a new {@link GdchCredentials} with an
+   * audience defaulted to the endpoint if the audience is not manually passed. This test confirms
+   * that a new credential is created from the context and can be refreshed
+   *
+   * @throws IOException
+   */
+  @Test
+  public void testCreateClient_withGdchCredentialAndNoAudience() throws IOException {
+    context = ClientContext.create(settings);
+    stubSettings = EchoStubSettings.newBuilder(context).build();
+    client = EchoClient.create(stubSettings.createStub());
+    Credentials fromContext = context.getCredentials();
+    Credentials fromClient = initialCredentials;
+    assertNotSame(fromContext, fromClient);
+    NullPointerException expectedEx =
+        assertThrows(NullPointerException.class, () -> initialCredentials.refresh());
+    assertTrue(
+        expectedEx.getMessage().contains("Audience are not configured for GDCH service account"));
+    Exception unexpected = null;
+    try {
+      registerCredential(fromContext);
+      ((GdchCredentials) fromContext).refreshAccessToken();
+    } catch (Exception ex) {
+      unexpected = ex;
+    }
+    assertNull(unexpected);
```

Comment:
Usually we don't have to specifically verify that no exception being thrown. The test case would fail naturally if there is an exception thrown, just `((GdchCredentials) fromContext).refreshAccessToken();` with a comment is more readable.

Reply:
added a simple call with comment instead. Thanks!